sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullFunctions.scala

```diff
@@ -25,13 +25,15 @@ import org.apache.spark.sql.types._
 
 
 /**
- * An expression that is evaluated to the first non-null input.
+ * An expression that is evaluated to the first non-null and non-NaN input.
```

Comment:
This change may be not related to this PR, will it change the behavior of sum() or avg()?

If yes, I think we have not decide how to do sum() on NaN yet. Can you leave this change out of this PR?